x-pack/plugins/enterprise_search/public/applications/app_search/components/relevance_tuning/types.ts

```diff
@@ -29,26 +29,38 @@ export enum BoostOperation {
   Add = 'add',
   Multiply = 'multiply',
 }
-
-export interface BaseBoost {
+export interface Boost {
+  type: BoostType;
   operation?: BoostOperation;
   function?: BoostFunction;
+  newBoost?: boolean;
+  center?: string | number;
+  factor: number;
+  value?: string[];
```

Comment:
(just curious, feel free to stop me if I'm missing something)

Do we really need this base `Boost` type any longer if the more specific Boost types below are overriding `value`, `function`, and ` operation` in any case? 🤷

Reply:
I think the reason it's nice to have that is that is for when we're passing a boost around in the general sense, and don't really care what type of boost it is.

Like in a reducer I can accept any `Boost` as a prop and check its `newBoost` prop. I don't really care what type of Boost it is, I just care that it's a type of `Boost`.

It's kind of like a classic inheritance thing, right?

Reply:
Gotcha, totally makes sense! Thanks for taking the time to explain!